components/esp_wifi_remote/wifi_remote_net.c

```diff
@@ -0,0 +1,90 @@
+/*
+ * SPDX-FileCopyrightText: 2023-2024 Espressif Systems (Shanghai) CO LTD
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <esp_private/wifi.h>
+#include "esp_err.h"
+#include "esp_wifi_remote.h"
+#include "esp_log.h"
+
+#define CHANNELS 2
+
+static esp_remote_channel_tx_fn_t s_tx_cb[CHANNELS];
+static esp_remote_channel_t s_channel[CHANNELS];
+static wifi_rxcb_t s_rx_fn[CHANNELS];
+
+esp_err_t esp_wifi_remote_channel_rx(void *h, void *buffer, void *buff_to_free, size_t len)
+{
+    assert(h);
+    if (h == s_channel[0]) {
+        assert(s_rx_fn[0]);
+        return s_rx_fn[0](buffer, len, buff_to_free);
+    }
+    if (h == s_channel[1]) {
+        assert(s_rx_fn[1]);
+        return s_rx_fn[1](buffer, len, buff_to_free);
+    }
+    return ESP_FAIL;
+}
+
+esp_err_t esp_wifi_remote_channel_set(wifi_interface_t ifx, void *h, esp_remote_channel_tx_fn_t tx_cb)
+{
+    if (ifx == WIFI_IF_STA) {
+        s_channel[0] = h;
+        s_tx_cb[0] = tx_cb;
+        return ESP_OK;
+    }
+    if (ifx == WIFI_IF_AP) {
+        s_channel[1] = h;
+        s_tx_cb[1] = tx_cb;
+        return ESP_OK;
+    }
+    return ESP_FAIL;
+}
+
+
+esp_err_t esp_wifi_internal_set_sta_ip(void)
```

Comment:
Should we add a LOG stating that this is not supported?

Reply:
I'll have to check where it is used. but will definitely add an error/debug log

Reply:
This is still a shortcut at the moment, this API is called from netif layers and notifies wifi library that we got an IP.
I've added a `TODO:...` comment, that we need to transfer this info to the slave device and all `esp_wifi_internal_set_sta_ip()` there